app/src/org/commcare/tasks/SaveToDiskTask.java

```diff
@@ -126,7 +127,7 @@ protected ResultAndError<SaveStatus> doTaskBackground(Void... nothing) {
             // Likely a user level issue, so send error to HQ as a app build error
             XPathErrorLogger.INSTANCE.logErrorToCurrentApp(cleanedMessage);
 
-            return new ResultAndError<>(SaveStatus.SAVE_ERROR, cleanedMessage);
+            return new ResultAndError<>(SaveStatus.SAVE_UNRECOVERABLE_ERROR, cleanedMessage);
```

Comment:
think all the places where we are throwing `SaveStatus.SAVE_ERROR` above should be `SaveStatus.SAVE_UNRECOVERABLE_ERROR` with this change.